qlog/qlog_dir.go

```diff
@@ -0,0 +1,49 @@
+package qlog
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/quic-go/quic-go/internal/utils"
+	"github.com/quic-go/quic-go/logging"
+)
+
+// DefaultTracer creates a qlog file in the qlog directory specified by the QLOGDIR environment variable.
+// File names are <odcid>_<perspective>.qlog.
+// Returns nil if QLOGDIR is not set.
+func DefaultTracer(_ context.Context, p logging.Perspective, connID logging.ConnectionID) *logging.ConnectionTracer {
+	var label string
+	switch p {
+	case logging.PerspectiveClient:
+		label = "client"
+	case logging.PerspectiveServer:
+		label = "server"
+	}
+	return qlogDirTracer(p, connID, label)
+}
+
+// qlogDirTracer creates a qlog file in the qlog directory specified by the QLOGDIR environment variable.
+// File names are <odcid>_<label>.qlog.
+// Returns nil if QLOGDIR is not set.
+func qlogDirTracer(p logging.Perspective, connID logging.ConnectionID, label string) *logging.ConnectionTracer {
+	qlogDir := os.Getenv("QLOGDIR")
+	if qlogDir == "" {
+		return nil
+	}
+	if _, err := os.Stat(qlogDir); os.IsNotExist(err) {
```

Comment:
1. Should we handle non-nil errors that are not `os.IsNotExist` errors?
2. The documentation for `os.IsNotExist` says we should use `errors.Is(err, fs.ErrNotExist)`.

Reply:
I'll do this in a follow-up PR. I might also change the file name a bit.

Reply:
Sounds good to me!